tests/unittests/sources/helpers/test_netlink.py

```diff
@@ -75,18 +77,18 @@ def test_netlink_read_timeout(self, m_select, m_socket):
         m_select.return_value = [], None, None
         data = read_netlink_socket(m_socket, 1)
         m_select.assert_called_with([m_socket], [], [], 1)
-        self.assertEqual(m_socket.recv.call_count, 0)
-        self.assertIsNone(data)
+        assert m_socket.recv.call_count == 0
+        assert data is None
 
     def test_read_invalid_socket(self):
         """read_netlink_socket raises assert error if socket is invalid"""
         socket = None
-        with self.assertRaises(AssertionError) as context:
+        with pytest.raises(AssertionError) as excinfo:
```

Comment:
Same here too allowing us to drop that separate str cast and assert below
```suggestion
        with pytest.raises(
            AssertionError, match="netlink socket is none"
        ):
```